src/libertem/io/dataset/hdf5.py

```diff
@@ -253,58 +264,159 @@ def get_diagnostics(self):
     def get_partitions(self):
         ds_shape = Shape(self.shape, sig_dims=self.sig_dims)
         ds_slice = Slice(origin=[0] * len(self.shape), shape=ds_shape)
+        target_size = self.target_size
+        if target_size is None:
+            if self._compression is None:
+                target_size = 512*1024*1024
+            else:
+                target_size = 256*1024*1024
         partition_shape = self.partition_shape(
-            target_size=self.target_size,
+            target_size=target_size,
             dtype=self.dtype,
         ) + tuple(self.shape.sig)
+
+        # if the data is chunked in the navigation axes, choose a compatible
+        # partition size (even important for non-compressed data!)
+        chunks = self._chunks
+        weird_chunks = chunks is not None and len(chunks) == 4 and chunks[0] > 1
```

Comment:
Why `len(chunks) == 4`? That would only be 4D STEM, right? What about data with other dimensions? How is the chunking returned if, for example, only the navigation dimension or only the signal dimension is chunked?

Reply:
> Why len(chunks) == 4? That would only be 4D STEM, right? What about data with other dimensions?

Yes, it's right now specialized for the 4D case. The ND case would cause more complexity, like this I can just write the solution down. I can generalize this to ND with more code, for sure.

> How is the chunking returned if, for example, only the navigation dimension or only the signal dimension is chunked?

The number of chunking dimensions always match the dataset shape.

Reply:
This is now generalized to the ND case, not yet tested/benchmarked with realistic data.